components/camel-jetty9/src/main/java/org/apache/camel/component/jetty9/JettyContentExchange9.java

```diff
@@ -214,14 +220,28 @@ public void onFailure(Request request, Throwable failure) {
             }
 
         };
-        BufferingResponseListener responseListener = new BufferingResponseListener() {
+
+        InputStreamResponseListener responseListener = new InputStreamResponseListener() {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+            @Override
+            public void onContent(Response response, ByteBuffer content, Callback callback) {
+                if (response.getStatus() != SC_OK) {
```

Comment:
I wasn't sure what to do in the case of a non-200 status here.  Treating this as a failure (in this case calling doTaskCompleted()) caused a bunch of other tests to fail.  Should I treat this as a failure and update the other tests, or just ignore it?